lib/bundler/shared_helpers.rb

```diff
@@ -221,12 +221,13 @@ def md5_available?
   private
 
     def validate_bundle_path
-      return unless Bundler.bundle_path.to_s.include?(File::PATH_SEPARATOR)
-      message = "Your bundle path contains a '#{File::PATH_SEPARATOR}', " \
+      path_separator = Gem.respond_to?(:path_separator) ? Gem.path_separator : File::PATH_SEPARATOR
```

Comment:
this conditional belongs in a method in `RubygemsIntegration`

Reply:
Done (via polymorphism instead of a conditional)